app/controllers/orders/conversation.js

```diff
@@ -1,3 +1,76 @@
-import detail from "./detail";
+import Ember from "ember";
+import config from '../../config/environment';
+import detail from './detail';
 
-export default detail.extend();
+export default detail.extend({
+  body: '',
+  isPrivate: false,
+  backLinkPath: "",
+  isMobileApp: config.cordova.enabled,
+  itemIdforHistoryRoute: null,
+  organisationIdforHistoryRoute: null,
+  i18n: Ember.inject.service(),
+  sortProperties: [
+    "createdAt: asc"
+  ],
+  model: null,
+  noMessage: Ember.computed.empty("model.messages"),
+
+  displayChatNote: Ember.computed('noMessage', 'disabled', function () {
+    return this.get("noMessage") && !this.get("disabled");
+  }),
+
+  sortedMessages: Ember.computed.sort("model.messages", "sortProperties"),
+
+  groupedMessages: Ember.computed("sortedMessages", function () {
+    return this.groupBy(this.get("sortedMessages"), "createdDate");
+  }),
+
+  groupBy: function (content, key) {
+    var result = [];
+    var object, value;
+
+    content.forEach(function (item) {
+      value = item.get ? item.get(key) : item[key];
```

Comment:
No need for this condition , you can do simply by `Ember.get(item, key)`

Reply:
sure, thanks @abulsayyad123